pybamm/expression_tree/symbol.py

```diff
@@ -952,21 +952,47 @@ def to_casadi(
         """
         return pybamm.CasadiConverter(casadi_symbols).convert(self, t, y, y_dot, inputs)
 
-    def create_copy(self):
+    def _children_for_copying(self, children: list[Symbol] | None = None) -> Symbol:
+        """
+        Gets existing children for a symbol being copied if they aren't provided.
+        """
+        if children is None:
+            children = [child.create_copy() for child in self.children]
+        return children
+
+    def create_copy(
```

Comment:
what is the difference between `create_copy` and `new_copy`? The docs say that `new_copy` adds attributes, but not sure what this means?

Reply:
new_copy used the print_name setter to ensure the print name was run through the pretty-print functions. No tests broke when I took this out, so I've deprecated new_copy in favour of just using `create_copy()` (to avoid confusion with `model.new_copy()`)